iib/web/api_v1.py

```diff
@@ -104,6 +104,30 @@ def _get_safe_args(args, payload):
     return safe_args
 
 
+def _get_unique_bundles(bundles):
+    """
+    Return list with unique bundles.
+
+    :param list bundles: bundles given in payload from original request
+    :return: list of unique bundles preserving order (python 3.6+)
+    :rtype: list
+    """
+    if not bundles:
+        return bundles
+
+    unique_bundles = list(dict.fromkeys(bundles).keys())
```

Comment:
Thank yo for asking, 

It is filtering out duplicates with preserving order. And it is similar to `list(set())`. 

**Explanation:**
Python type dictionary `dict` is preserving order of inserted keys since Python 3.6. IIB uses python 3.6 and newer. It does not support lower due to heavily used f-strings. 
Keys in dictionary are behaving as a `set()`. You can not have same key twice. Therefore I am creating dictionary where keys are taken from `bundles` (which is list of strings) using `dict.fromkeys()` method. This will create dictionary with unique keys with same order as it is in `bundles`. After that I take all the keys from this dictionary using `.keys()` and converting them to list using `list()`, because bundles should be type of list of strings. 

Is it OK? Or should I try to explain it better?

Reply:
Thank you it is perfect. I didn't exactly know how the bundles look but it make sense now. :) i would suggest to even copy this text as a comment there.

Reply:
point taken, comment added